# **5. Create Your Portfolio**

Rincian Tugas:
1. Carilah sebuah dataset (dari kaggle ataupun website lainnya)
2. Lakukan Exploratory Data Analysis pada dataset tersebut.
  - EDA yang dilakukan WAJIB melakukan `Handling Missing Value dan Duplicate`
  - Proses EDA yang lain bersifat opsional (e.g. *handling outlier*, *encoding*)
3. Buat portfolio dalam bentuk Power Point
4. Upload di LinkedIn dan tag dibimbing.id

In [ ]:
from google.colab import files
uplouded = files.upload()

import pandas as pd
df = pd.read_csv('Vehicle Price (1).csv')
df.head()

Saving Vehicle Price (1).csv to Vehicle Price (1) (1).csv


name  \
0     2024 Jeep Wagoneer Series II   
1  2024 Jeep Grand Cherokee Laredo   
2         2024 GMC Yukon XL Denali   
3       2023 Dodge Durango Pursuit   
4            2024 RAM 3500 Laramie   

                                         description   make           model  \
0  \n      \n        Heated Leather Seats, Nav Sy...   Jeep        Wagoneer   
1  Al West is committed to offering every custome...   Jeep  Grand Cherokee   
2                                                NaN    GMC        Yukon XL   
3  White Knuckle Clearcoat 2023 Dodge Durango Pur...  Dodge         Durango   
4  \n      \n        2024 Ram 3500 Laramie Billet...    RAM            3500   

   year    price                                             engine  \
0  2024  74600.0                            24V GDI DOHC Twin Turbo   
1  2024  50170.0                                                OHV   
2  2024  96410.0  6.2L V-8 gasoline direct injection, variable v...   
3  2023  46835.0                                       16V MPFI OHV   
4  2024  81663.0                           24V DDI OHV Turbo Diesel   

   cylinders      fuel  mileage       transmission       trim          body  \
0        6.0  Gasoline     10.0  8-Speed Automatic  Series II           SUV   
1        6.0  Gasoline      1.0  8-Speed Automatic     Laredo           SUV   
2        8.0  Gasoline      0.0          Automatic     Denali           SUV   
3        8.0  Gasoline     32.0  8-Speed Automatic    Pursuit           SUV   
4        6.0    Diesel     10.0  6-Speed Automatic    Laramie  Pickup Truck   

   doors           exterior_color    interior_color        drivetrain  
0    4.0                    White      Global Black  Four-wheel Drive  
1    4.0                 Metallic      Global Black  Four-wheel Drive  
2    4.0             Summit White  Teak/Light Shale  Four-wheel Drive  
3    4.0  White Knuckle Clearcoat             Black   All-wheel Drive  
4    4.0                   Silver             Black  Four-wheel Drive

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1002 entries, 0 to 1001
Data columns (total 17 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   name            1002 non-null   object 
 1   description     946 non-null    object 
 2   make            1002 non-null   object 
 3   model           1002 non-null   object 
 4   year            1002 non-null   int64  
 5   price           979 non-null    float64
 6   engine          1000 non-null   object 
 7   cylinders       897 non-null    float64
 8   fuel            995 non-null    object 
 9   mileage         968 non-null    float64
 10  transmission    1000 non-null   object 
 11  trim            1001 non-null   object 
 12  body            999 non-null    object 
 13  doors           995 non-null    float64
 14  exterior_color  997 non-null    object 
 15  interior_color  964 non-null    object 
 16  drivetrain      1002 non-null   object 
dtypes: float64(4), int64(1), object(1

In [ ]:
# cek missing value
df.isna().sum()

name                0
description        56
make                0
model               0
year                0
price              23
engine              2
cylinders         105
fuel                7
mileage            34
transmission        2
trim                1
body                3
doors               7
exterior_color      5
interior_color     38
drivetrain          0
dtype: int64

**Observasi** :
Dari informasi yang didapat pada data yang terdapat beberapa data yang hilang pada kolom description, price, engine, cylinders, fuel, mileage, transmission, trim, body, doors, exterior_color, dan interior_color

In [ ]:
# cek statiscial summary
df.describe()

year          price   cylinders      mileage       doors
count  1002.000000     979.000000  897.000000   968.000000  995.000000
mean   2023.916168   50202.985700    4.975474    69.033058    3.943719
std       0.298109   18700.392062    1.392526   507.435745    0.274409
min    2023.000000       0.000000    0.000000     0.000000    2.000000
25%    2024.000000   36600.000000    4.000000     4.000000    4.000000
50%    2024.000000   47165.000000    4.000000     8.000000    4.000000
75%    2024.000000   58919.500000    6.000000    13.000000    4.000000
max    2025.000000  195895.000000    8.000000  9711.000000    5.000000

In [ ]:
df['year'].describe()

count    1002.000000
mean     2023.916168
std         0.298109
min      2023.000000
25%      2024.000000
50%      2024.000000
75%      2024.000000
max      2025.000000
Name: year, dtype: float64

**Observasi Kolom** *year* :
1. **mean** (rata - rata tahun) adalah 2023.92, menujukkan bahwa mayoritas data berada di tahun 2024
2. **median (50%), Q1 (25%) dan Q3 (75%) seluruhnya bernilai 2024, artinya setidaknya 75% data berada ditahun 2024**, yang menunjukkan distribusi tahun yang sangat terpusat diangka tersebut.
3. **Standar Deviasi (std)** yang relatif kecil (0.29) menunjukkan bahwa variasi data *year* sangat rendah serta artinya data tersebut sangat stabil
4. tidak ditemukannya data yang hilang pada kolom year (count = 1002)
5. **2023** adalah tahun data paling awal sedangkan **2025** adalah data terbaru






In [ ]:
# untuk mengatasi missing value
for column in df.columns:
    if df[column].dtype == 'object':
        # Jika kolom bertipe object, isi dengan mode
        df[column].fillna(df[column].mode()[0], inplace=True)

/tmp/ipython-input-10-3865571342.py:5: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df[column].fillna(df[column].mode()[0], inplace=True)


In [ ]:
# mengecek kembali missing value
df.isna().sum()

name              0
description       0
make              0
model             0
year              0
price             0
engine            0
cylinders         0
fuel              0
mileage           0
transmission      0
trim              0
body              0
doors             0
exterior_color    0
interior_color    0
drivetrain        0
dtype: int64

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1002 entries, 0 to 1001
Data columns (total 17 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   name            1002 non-null   object 
 1   description     1002 non-null   object 
 2   make            1002 non-null   object 
 3   model           1002 non-null   object 
 4   year            1002 non-null   int64  
 5   price           1002 non-null   float64
 6   engine          1002 non-null   object 
 7   cylinders       1002 non-null   float64
 8   fuel            1002 non-null   object 
 9   mileage         1002 non-null   float64
 10  transmission    1002 non-null   object 
 11  trim            1002 non-null   object 
 12  body            1002 non-null   object 
 13  doors           1002 non-null   float64
 14  exterior_color  1002 non-null   object 
 15  interior_color  1002 non-null   object 
 16  drivetrain      1002 non-null   object 
dtypes: float64(4), int64(1), object(1

In [ ]:
check_duplicate = df.duplicated().sum()

print(f"Jumlah data yang duplikat = {check_duplicate}")

Jumlah data yang duplikat = 24


In [ ]:
# Handling duplicate
df = df.drop_duplicates()

In [ ]:
# mengecek duplicate setelah di-handle
handle_duplicate = df.duplicated().sum()

print(f"Jumlah data yang duplikat = {handle_duplicate}")

Jumlah data yang duplikat = 0


In [ ]:
# Handling Outlier
# cek kolom numerik
df.select_dtypes(include=['float64', 'int64']).columns

Index(['year', 'price', 'cylinders', 'mileage', 'doors'], dtype='object')

In [ ]:
Q1 = df['price'].quantile(0.25)
Q3 = df['price'].quantile(0.75)
IQR = Q3 - Q1

batas_bawah = Q1 - 1.5 * IQR
batas_atas = Q3 + 1.5 * IQR

# melihat data outlier
outlier = df[(df['price'] < batas_bawah) | (df['price'] > batas_atas)]
outlier
# mengecek jumlah outlier
print("Jumlah outlier pada kolom price:", len(outlier))

Jumlah outlier pada kolom price: 26


In [ ]:
# menangani Outlier pada kolom price
df = df[(df['price'] >= batas_bawah) & (df['price'] <= batas_atas)]

In [ ]:
# mengecek kembali jumlah outlier pada kolom price
outlier = df[(df['price'] < batas_bawah) | (df['price'] > batas_atas)]
print("Jumlah outlier pada kolom price:", len(outlier))

Jumlah outlier pada kolom price: 0


In [ ]:
# Encoding
# mengecek kolom kategorikal
df.select_dtypes(include=['object']).columns

Index(['name', 'description', 'make', 'model', 'engine', 'fuel',
       'transmission', 'trim', 'body', 'exterior_color', 'interior_color',
       'drivetrain'],
      dtype='object')

In [ ]:
df = pd.get_dummies(df, columns=['name', 'description', 'make', 'model', 'engine', 'fuel',
       'transmission', 'trim', 'body', 'exterior_color', 'interior_color',
       'drivetrain'], drop_first=True)

In [ ]:
# mengecek kembali
df.select_dtypes(include=['object']).columns

Index([], dtype='object')